main/components/UploadDocumentsForm.tsx

```diff
@@ -11,27 +13,93 @@ export function UploadDocumentsForm() {
     setFile(e.target.files[0]);
   };
 
+  const convertXlsx = (file: any) => {
+    return new Promise((resolve, reject) => {
+      const reader = new FileReader();
+  
+      reader.onload = (e) => {
+        try {
+          const data = e.target.result;
+          const workbook = XLSX.read(data, { type: 'binary' });
+  
+          let allCsvData = '';
+  
+          const promises = workbook.SheetNames.map(async (sheetName, index) => {
+            if (index > 0) {
+              // Add a newline between sheets
+              allCsvData += '\n';
+            }
+  
+            const formData = new FormData();
+            const csv = XLSX.utils.sheet_to_csv(workbook.Sheets[sheetName]);
+            await formData.append('file', new Blob([csv], { type: 'text/csv' }), file.name + '_converted.csv');
+  
+            await axios.post('/api/retrieval/file_ingest', formData)
+              .then(res => {
+                console.log(res);
+              })
+              .catch(er => {
+                console.log(er);
+              });
+  
+            allCsvData += csv;
+          });
+  
+          Promise.all(promises)
+            .then(() => {
+              resolve(allCsvData);
+            })
+            .catch((error) => {
+              reject(error);
+            });
+        } catch (error) {
+          reject(error);
+        }
+      };
+  
+      reader.onerror = (error) => {
+        reject(error);
+      };
+  
+      reader.readAsBinaryString(file);
+    });
+  };
   const ingest = async (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     setIsLoading(true);
-    const formData = new FormData()
-    formData.append('file', file);
-    formData.append('chunkSize', String(process.env.NEXT_PUBLIC_CHUNK_SIZE));
-    axios.post('/api/retrieval/file_ingest', formData)
-    .then( res => {
-      console.log(res);
-      alert("Document uploaded successfully");
-      router.push("/retrieval");
-      setFile(null);
-      setIsLoading(false);
-    })
-    .catch(er => {
-      console.log(er);
-      setFile(null);
-      setIsLoading(false);
-    });
+    
+    const fileExtension = path.extname(file.name);
+    if (fileExtension === '.xlsx' || fileExtension === '.xls') {
+      try {
+        const csvData = await convertXlsx(file);
+        await alert('Document uploaded successfully');
+        router.push('/retrieval');
+        setFile(null);
+      } catch (error) {
+        console.log('Error converting to csv:', error);
+      } finally {
+        setIsLoading(false);
+      }
```

Comment:
Can we and should we do this conversion from Excel to CSV in the backend? I think keeping all the file processing in the backend maybe cleaner. But it's fine if it ends up needing more work than just moving this code to app/api/retrieval/file_ingest

Reply:
when xlsx convert into csv only in backend(I tried it), the row and column does not match if they have different sheets, for instance the second sheet of data will use column and row information from the first sheet which is not the case. There might be better way to handle this, but as of now this is fully functional thus pushed.

Reply:
I see. it's fine then